src/btree/bt_io.c

```diff
@@ -35,12 +35,16 @@ __wt_bt_read(WT_SESSION_IMPL *session, WT_ITEM *buf, const uint8_t *addr, size_t
      * into the caller's buffer. Else, read directly into the caller's buffer.
      */
     if (btree->compressor == NULL && btree->kencryptor == NULL) {
-        WT_RET(bm->read(bm, session, buf, addr, addr_size));
+        WT_WITH_BUCKET_STORAGE(
+          btree->bstorage, session, { ret = bm->read(bm, session, buf, addr, addr_size); });
+        WT_RET(ret);
         dsk = buf->data;
         ip = NULL;
     } else {
         WT_RET(__wt_scr_alloc(session, 0, &tmp));
-        WT_ERR(bm->read(bm, session, tmp, addr, addr_size));
+        WT_WITH_BUCKET_STORAGE(
+          btree->bstorage, session, { ret = bm->read(bm, session, tmp, addr, addr_size); });
+        WT_ERR(ret);
```

Comment:
Yes, we're going to collide a bit here because the bstorage is moving to `tiered->bstorage`.